### PR TITLE
Add documentation for writing new device

### PR DESCRIPTION
**Pull request by albi3ro**
We currently have a page to describe how to develop a new plugin. With the addition of the new Device API, we need to update that page.

### CODE REVIEW

File: doc/development/plugins.rst [hunk 1]
```diff
@@ -1,395 +1,134 @@
...
-  *  ``'supports_tracker'`` (*bool*): ``True`` if it has a device tracker attribute and updates information with
-     it.
-
-  Some capabilities are queried by PennyLane core to make decisions on how to best run computations, others are used
-  by external apps built on top of the device ecosystem.
-
-  To find out which capabilities are (possibly automatically) defined for your device ``dev = device('my.device')``,
-  check the output of ``dev.capabilities()``.
-
-Adding arguments to your device
```
Comment (trbromley) [inline doc/development/plugins.rst:162]: Same for this section?

File: doc/development/plugins.rst [hunk 2]
```diff
@@ -1,395 +1,134 @@
...
-
-
-Wire handling
--------------
-
-PennyLane uses the :class:`~.wires.Wires` class for the internal representation of wires. :class:`~.wires.Wires`
-inherits from Python's ``Sequence``, and represents an ordered set of unique wire labels.
-Indexing a ``Wires`` instance will return another ``Wires`` instance of length one.
-The ``labels`` attribute stores a tuple of the wire labels.
+        def execute(self, circuits: "QuantumTape_or_Batch", execution_config: "ExecutionConfig" = DefaultExecutionConfig):
```
Comment (trbromley) [inline doc/development/plugins.rst:28]: > QuantumTape_or_Batch

Is this in place of a `Union`?

File: doc/development/plugins.rst [hunk 3]
```diff
@@ -1,395 +1,134 @@
...
+(0.2, -0.052, -0.014)
+
+The first number ``0.2`` is calculated with 5 shots, the second ``-0.052`` is calculated with 500 shots, and
+``-0.014`` is calculated with 1000 shots.  All 1,505 shots can be requested together in one batch, but the post
+processing into the expecation value is done with shots ``0:5``, ``5:505``, and ``505:1505`` respectively.
+
+Wires
+-----
+
+Devices can now either:
```
Comment (trbromley) [inline doc/development/plugins.rst:87]:
```suggestion
Devices can now either:

```

File: doc/development/plugins.rst [hunk 4]
```diff
@@ -1,395 +1,134 @@
...
+1) Strictly use wires provided by the user on initialization ``device(name, wires=wires)``
+2) Infer the number and ordering of wires provided by the submitted circuit.
```
Comment (trbromley) [inline doc/development/plugins.rst:89]:
```suggestion
1. Strictly use wires provided by the user on initialization ``device(name, wires=wires)``
2. Infer the number and ordering of wires provided by the submitted circuit.
```

**Review by trbromley — COMMENTED** [review]
Thanks @albi3ro!

File: doc/development/plugins.rst (hunk 2 above)
Comment (albi3ro) [reply doc/development/plugins.rst:28]: I probably should just right out `Union[QuantumTape, Sequence[QuantumTape]]`.

File: doc/development/plugins.rst [hunk 5]
```diff
@@ -1,395 +1,134 @@
...
-* Each plugin may provide one (or more) devices, that are accessible directly through
-  PennyLane, as well as any additional private functions or classes.
-
-* Depending on the scope of the plugin, you may wish to provide additional (custom)
-  quantum operations and observables that the user can import.
-
-.. important::
-
-    In your plugin module, **standard NumPy** (*not* the wrapped Autograd version of NumPy)
-    should be imported in all places (i.e., ``import numpy as np``).
```
[the remark this answers is not in the document]
Comment (albi3ro) [reply doc/development/plugins.rst:33]: I started by removing everything and then explicitly copying certain things back over. So still copying things over.

File: doc/development/plugins.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (albi3ro) [reply doc/development/plugins.rst:28]: I have a section on measurement processes. I can probably group that into a section with the python qubit simulation utilities as well.

**Comment by albi3ro** [comment]
[sc-45714]

**Review by trbromley — COMMENTED** [review]
Thanks @albi3ro, I've posted a few more thoughts from during hack week.

**Comment by albi3ro** [comment]
Closed in favor of #5783

**Pull request closed without merging**